heron/common/src/cpp/network/client.cpp

```diff
@@ -181,8 +181,10 @@ void Client::OnNewPacket(IncomingPacket* _ipkt) {
 
   if (_ipkt->UnPackString(&typname) != 0) {
     Connection* conn = static_cast<Connection*>(conn_);
-    LOG(FATAL) << "UnPackString failed from connection " << conn << " from hostport "
+    LOG(ERROR) << "UnPackString failed from connection " << conn << " from hostport "
```

Comment:
On the one hand, not sure whether it would affect our semantics: originally it is fail-fast but now it is dropping. On the other hand, we would also need to export metrics for this behavior; otherwise, it is hard to trace and debug.

Also, did we really witness this data corruption, since protobuf is pretty neat.